src/tt.cpp

```diff
@@ -75,6 +91,14 @@ void TranspositionTable::resize(size_t mbSize) {
       exit(EXIT_FAILURE);
   }
 
+#ifdef USE_MADVISE_HUGEPAGE
+  // Request huge pages in case aligned_alloc() didn't already provide them. We
+  // don't care if this call fails, and the call may fail if transparent huge
+  // pages are not enable at all. Note that MADV_HUGEPAGE may not be available
+  // on every platform that supports madvise().
+  madvise(mem, allocSize, MADV_HUGEPAGE);
```

Comment:
Would
```C++
madvise(mem, allocSize, MADV_HUGEPAGE | MADV_RANDOM);
```
be even better?

Reply:
A reasonable consideration. However:

- The advice parameter is not actually a bitfield, so you can't use bitwise or.
- Since the TT hash is always supposed to be resident after the initial memset() to clear it, MADV_RANDOM doesn't actually do anything to my knowledge. Those MADV_RANDOM/MADV_SEQUENTIAL are supposed to control prefetching (i.e., file read-ahead) of swapped-out pages or pages in memory-mapped files. MADV_RANDOM might theoretically make sense for the Syzygy files to control resident memory bloat, but it needs a separate study.

The one advice that I would have liked to use here is MADV_WILLNEED to make pages resident without page faulting on the initial memset(). This should speed up hash init even more. Unfortunately, it doesn't seem to do anything for anonymous mappings (i.e., plain memory not mapped to files).